From 5fcfff7c32086ad99b8177d5e71e5bc24ae280e9 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:01:58 -0400
Subject: [ethernet] mlx4: Use is_kdump_kernel() to detect kdump kernel

Message-id: <32ad54cb031ca4c67bf55c534fd802f0a8102c4f.1410425016.git.avadai@redhat.com>
Patchwork-id: 93723
O-Subject: [Patch RHEL7.1 V1 182/197] net/mlx4: Use is_kdump_kernel() to detect kdump kernel
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Amir Vadai <amirv@mellanox.com>

Use is_kdump_kernel() to detect kdump kernel, instead of reset_devices.

Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 48ea526a6877d605c961aa37fae33f3227b29424)

Bugzilla: 1121698
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/mlx4/device.h b/include/linux/mlx4/device.h
index c11e895..b28026e 100644
--- a/include/linux/mlx4/device.h
+++ b/include/linux/mlx4/device.h
@@ -38,6 +38,7 @@
 #include <linux/completion.h>
 #include <linux/radix-tree.h>
 #include <linux/cpu_rmap.h>
+#include <linux/crash_dump.h>
 
 #include <linux/atomic.h>
 
@@ -1275,7 +1276,7 @@ int mlx4_mr_rereg_mem_write(struct mlx4_dev *dev, struct mlx4_mr *mr,
 /* Returns true if running in low memory profile (kdump kernel) */
 static inline bool mlx4_low_memory_profile(void)
 {
- return reset_devices;
+ return is_kdump_kernel();
 }
 
 #endif /* MLX4_DEVICE_H */
-- 
1.7.1